code/game/machinery/prisongate.dm

```diff
@@ -0,0 +1,81 @@
+#define SPAM_CD 3 SECONDS
+
+/obj/machinery/prisongate
+	name = "prison gate scanner"
+	desc = "A hardlight gate with an ID scanner attached to the side. Good at deterring even the most persistent temporarily embarrassed employee."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "prisongate_on"
+	/// roughly the same health/armor as an airlock
+	max_integrity = 450
+	armor = list(MELEE = 30, BULLET = 30, LASER = 20, ENERGY = 20, BOMB = 10, BIO = 100, FIRE = 80, ACID = 70)
+	use_power = IDLE_POWER_USE
+	power_channel = AREA_USAGE_EQUIP
+	idle_power_usage = 5
+	active_power_usage = 30
+	anchored = TRUE
+	/// dictates whether the gate barrier is up or not
+	var/gate_active = TRUE
+	COOLDOWN_DECLARE(spam_cooldown_time)
+
+/obj/machinery/prisongate/power_change()
+	. = ..()
+	if(!powered())
+		visible_message(span_notice("[src] momentarily flickers before the hardlight barrier loses cohesion and dissipates into thin air!"))
+		gate_active = FALSE
+		flick("prisongate_turningoff", src)
+		icon_state = "prisongate_off"
+	else
+		gate_active = TRUE
+		visible_message(span_notice("[src] whirrs back to life as its hardlight barrier fills the space between it."))
+		flick("prisongate_turningon", src)
+		icon_state = "prisongate_on"
+
+/obj/machinery/prisongate/CanAllowThrough(atom/movable/gate_toucher, border_dir)
+	. = ..()
+	if(!iscarbon(gate_toucher))
+		if(isstructure(gate_toucher))
```

Comment:
early return this. so if it isn't a structure, return TRUE. then unindent the following block